sklearn/cluster/_dbscan.py

```diff
@@ -298,6 +298,17 @@ class DBSCAN(ClusterMixin, BaseEstimator):
     DBSCAN(eps=3, min_samples=2)
     """
 
+    _parameter_constraints = {
+        "eps": [Interval(Real, 0.0, None, closed="neither")],
+        "min_samples": [Interval(Integral, 1, None, closed="left")],
+        "metric": [str, callable],
```

Comment:
I believe `DBSCAN` accepts exactly every metric that  `NearestNeighbors` accepts (as it is used in calculating core-points) and this is available as `sklearn.metrics.pairwise._VALID_METRICS`

```suggestion
        "metric": [StrOptions(_VALID_METRICS), callable],
```

Reply:
It was a really nice suggestion, but there were many errors in the tests:
```
ValueError: The 'options' parameter of StrOptions.__init__ must be an instance of 'set'. Got ['euclidean', 'l2', 'l1', 'manhattan', 'cityblock', 'braycurtis', 'canberra', 'chebyshev', 'correlation', 'cosine', 'dice', 'hamming', 'jaccard', 'kulsinski', 'mahalanobis', 'matching', 'minkowski', 'rogerstanimoto', 'russellrao', 'seuclidean', 'sokalmichener', 'sokalsneath', 'sqeuclidean', 'yule', 'wminkowski', 'nan_euclidean', 'haversine'] instead.
```
Therefore I proceed to revert the suggested changes.

Reply:
Suggestion applied. I had to transform `_VALID_METRICS` to a set, and add `"precomputed"` to the set of valid values so that the tests are green.